validate_sample.py

```diff
@@ -0,0 +1,339 @@
+                  f'Value needs to be numeric with max precision of {max_precision} and max scale of {max_scale}')
+            error_count += 1
+
+
+def validate_oa(line_number, sample_row):
+    column = 'OA'
+    maximum_length = 9
+    if _check_mandatory_value_exists(line_number, column, mandatory=True, sample_row=sample_row):
+        _check_length(column, sample_row[column], line_number, maximum_length)
+
+
+def validate_lsoa(line_number, sample_row):
+    column = 'LSOA'
+    maximum_length = 9
+    if _check_mandatory_value_exists(line_number, column, mandatory=True, sample_row=sample_row):
+        _check_length(column, sample_row[column], line_number, maximum_length)
+
+
+def validate_msoa(line_number, sample_row):
+    column = 'MSOA'
+    maximum_length = 9
+    if _check_mandatory_value_exists(line_number, column, mandatory=True, sample_row=sample_row):
+        _check_length(column, sample_row[column], line_number, maximum_length)
+
+
+def validate_lad(line_number, sample_row):
+    column = 'LAD'
+    maximum_length = 9
+    if _check_mandatory_value_exists(line_number, column, mandatory=True, sample_row=sample_row):
+        _check_length(column, sample_row[column], line_number, maximum_length)
+
+
+def validate_region(line_number, sample_row):
+    column = 'REGION'
+    maximum_length = 9
+    if _check_mandatory_value_exists(line_number, column, mandatory=True, sample_row=sample_row):
+        _check_length(column, sample_row[column], line_number, maximum_length)
+
+
+def validate_htc_willingness(line_number, sample_row):
+    global error_count
+    column = 'HTC_WILLINGNESS'
+    if _check_mandatory_value_exists(line_number, column, mandatory=True, sample_row=sample_row):
+        htc_willingness = sample_row[column]
+        if htc_willingness.isnumeric and len(htc_willingness) == 1:
+            return
+        print(f'Line {line_number}: {column} {htc_willingness} is not valid.')
+        error_count += 1
+
+
+def validate_htc_digital(line_number, sample_row):
+    global error_count
+    column = 'HTC_DIGITAL'
+    if _check_mandatory_value_exists(line_number, column, mandatory=True, sample_row=sample_row):
+        htc_digital = sample_row[column]
+        if htc_digital.isnumeric and len(htc_digital) == 1:
+            return
+        print(f'Line {line_number}: {column} {htc_digital} is not valid.')
+        error_count += 1
+
+
+def validate_fieldcordinator_id(line_number, sample_row):
+    column = 'FIELDCOORDINATOR_ID'
+    maximum_length = 7
+    if _check_mandatory_value_exists(line_number, column, mandatory=False, sample_row=sample_row):
+        _check_length(column, sample_row[column], line_number, maximum_length)
+
+
+def validate_fieldofficer_id(line_number, sample_row):
+    column = 'FIELDOFFICER_ID'
+    maximum_length = 10
+    if _check_mandatory_value_exists(line_number, column, mandatory=False, sample_row=sample_row):
```

Comment:
Field officer ID is mandatory in the spec. The sample generator script may need updating to populate it as well.
```suggestion
    if _check_mandatory_value_exists(line_number, column, mandatory=True, sample_row=sample_row):
```

Reply:
The spec is wrong (or at best misleading).  The field is only mandatory if it's gone out to field so for the purposes of this check, it'll be non-mandatory.  Future checks could check it against a treatment code soo see whether it's required but that's outside the scope of this work.